### PR TITLE
Bump moby to 0ede01237c9ab871f1b8db0364427407f3e46541

### PR DESCRIPTION
**Pull request by thaJeztah**
full diff: https://github.com/docker/docker/compare/079ed017b61eb819b8184b90013ce89465d3aaba...0ede01237c9ab871f1b8db0364427407f3e46541

Includes:

- https://github.com/moby/moby/pull/34899 [client] Remove duplicate NewClient functions
- https://github.com/moby/moby/pull/33702 Add API support for templated secrets and configs
- https://github.com/moby/moby/pull/36366 Adjust minimum API version for templated configs/secrets

### CODE REVIEW

**Comment by thaJeztah** [comment]
ping @cpuguy83 @vdemeester PTAL

File: vendor/github.com/docker/docker/api/common.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package api // import "github.com/docker/docker/api"
 // Common constants for daemon and client.
 const (
 	// DefaultVersion of Current REST API
-	DefaultVersion string = "1.36"
+	DefaultVersion string = "1.37"
```
Comment (thaJeztah) [inline vendor/github.com/docker/docker/api/common.go]: This was forgotten in the bump

**Comment by dnephin** [comment]
`client.ParseHost` was removed, so we need to update to use `client.ParseHostURL`. I can push a commit to this PR if you'd like.

**Comment by thaJeztah** [comment]
hmf, looks like changes are needed;

```
# github.com/docker/cli/cli/command
cli/command/cli.go:303:25: undefined: "github.com/docker/cli/vendor/github.com/docker/docker/client".ParseHost
```

**Comment by thaJeztah** [comment]
this is green now 👍

**Comment by thaJeztah** [comment]
oh, just now see you commented, looks like our comments just crossed

File: cli/command/cli.go [hunk 2]
```diff
@@ -300,12 +300,12 @@ func newHTTPClient(host string, tlsOptions *tlsconfig.Options) (*http.Client, er
...
 		}).DialContext,
 	}
-	proto, addr, _, err := client.ParseHost(host)
+	hostURL, err := client.ParseHostURL(host)
 	if err != nil {
 		return nil, err
 	}
 
-	sockets.ConfigureTransport(tr, proto, addr)
+	sockets.ConfigureTransport(tr, hostURL.Scheme, hostURL.Scheme)
```
Comment (thaJeztah) [inline cli/command/cli.go]: whoops, hold on

Comment (thaJeztah) [reply cli/command/cli.go]: .. generally it's better to save your changes before committing :joy:

**Pull request merged**